servers/mongo/server-mongo.js

```diff
@@ -11,24 +11,148 @@ app.use(express.urlencoded({ extended: true }));
 /*
 ROUTES
 */
-
 app.get('/qa/questions', (req,res) => {
+  let query = req.query;
+  let p_id = Number(query.product_id)
+  let page = 1;
+  let count = Number(query.count);
+
+  db.getAllQuestions(p_id, page, count)
+  .then(data => {
+    res.status(200).send(data)
+  })
+  .catch(err => {
+    console.log('error querying db', err)
+    res.status(400).send(typeof err)
+  })
+})
+
+app.get('/qa/questions/:question_id/answers', (req, res) => {
+  let query = req.query;
+  let params = req.params
+
+  let q_id = Number(params.question_id)
+  let page = Number(query.page)
+  let count = Number(query.count)
+
+  db.getAnswers(q_id, page, count).then(data => {
+    res.status(200).send(data)
+  })
+  .catch(err => {
+    console.log('error getting answers', err)
+    res.send(400)
+  })
+})
+
+app.post('/qa/questions', (req,res) => {
+  let query = req.query;
+
+  let body = query.body;
+  let name = query.name;
+  let email = query.email;
+  let p_id = query.product_id;
+
+  db.postQuestion(p_id, name, email, body)
+  .then(data => {
+    res.status(201).send(data)
+  })
+  .catch(err => {
+    console.log('error posting questions', err)
+  })
+})
+
+app.post('/qa/questions/:question_id/answers', (req, res) => {
+  let query = req.query;
+  let params = req.params
+
+  let q_id = Number(params.question_id);
+  let body = query.body;
+  let name = query.name;
+  let email = query.email;
+  let photos = query.photos || []
 
+  if(typeof query.photos === 'string') {
+    try {
+      photos = query.photos.split('').filter(l => {
+        return (l !== '[' && l !== ']')
+      }).join('').split(',')
+
+      console.log('photos', photos)
+
+      if (!photos) {
+        throw new Error('Incompatible data type for photos')
+      }
+    } catch (e) {
+      console.error(e)
```

Comment:
If photos data type is incompatible, do you intend not to proceed with the post request? Should you end request here with something like res.sendStatus?

Reply:
Ah good point, I will send a 400 response in the catch, thank you!